mava/utils/checkpointing.py

```diff
@@ -182,25 +175,21 @@ def restore_learner_state(
         # Dictionary of the restored learner state
         restored_learner_state_raw = restored_checkpoint["learner_state"]
 
-        # Restore the learner state type and check it matches the input type
-        restored_learner_state_type = locate(restored_checkpoint["type"])
-        assert restored_learner_state_type == type(unreplicated_input_learner_state)
-
-        # We base the new learner state on the input learner state
-        new_learner_state = copy(unreplicated_input_learner_state)
-
-        if restore_params:
-            new_learner_state = new_learner_state._replace(
-                params=Params(**FrozenDict(restored_learner_state_raw["params"])),
-            )
-
-        if restore_hstates and restored_learner_state_type == RNNLearnerState:
-            new_learner_state = typing.cast(RNNLearnerState, new_learner_state)  # for mypy
-            new_learner_state = new_learner_state._replace(
-                hstates=HiddenStates(**FrozenDict(restored_learner_state_raw["hstates"])),
-            )
+        # Restore params
+        if isinstance(input_params.actor_params, FrozenDict):
+            restored_params = Params(**FrozenDict(restored_learner_state_raw["params"]))
+        else:
+            restored_params = Params(**restored_learner_state_raw["params"])
```

Comment:
What's the difference here? Why would we store two different cases?

Reply:
Previously, it was necessary to unpack the params into a FrozenDict, and we used to only do:
```
Params(**FrozenDict(restored_learner_state_raw["params"]))
```
But @OmaymaMahjoub found that recently, the params are coming in as dicts. We don't really understand it, but I believe we're checking for both possibilities in case of a change later.

Reply:
Though perhaps it is better to figure out what is happening, and choose one approach 🤔

Reply:
From my side I think we can keep it as a sanity check

Reply:
Ok then can we just document this, basically put exact what you've written above - flax now returns dicts and we're being careful?

Reply:
the issue is that we are not 100% sure the reason was flax but I will try to downgrade it to the old version and see if it was the reason behind this

Reply:
Updates yeah flax 0.6.11 was having the networks as FrozenDict

Reply:
I updated the docs, feel free to resolve the conversation once you double check it